recipe/meta.yaml

```diff
@@ -1,99 +1,235 @@
+    - {{ cdt('libxxf86vm') }}            # [linux]
+    - {{ cdt('mesa-libgl-devel') }}      # [linux and not s390x]
+    - {{ cdt('mesa-libegl-devel') }}     # [linux and not ppc64le]
+    - {{ cdt('mesa-libgbm') }}           # [linux and not ppc64le]
+    - {{ cdt('mesa-dri-drivers') }}      # [linux]
     - {{ cdt('libx11-devel') }}          # [linux]
     - {{ cdt('libxau-devel') }}          # [linux]
     - {{ cdt('libxext-devel') }}         # [linux]
     - {{ cdt('libxrender-devel') }}      # [linux]
     - {{ cdt('xorg-x11-proto-devel') }}  # [linux]
+    - m2-msys2-runtime  # [win]
+    - pkg-config
+    - cmake
+    - {{ posix }}make
+    - ninja
+    - {{ posix }}patch
+    - perl 5.*
+    - gobject-introspection 1.*  # [not win]
   host:
-    - python >=3.6
-    - boost-cpp
-    - cairo  # [not win]
+    - python 3.9
+    - boost-cpp {{ boost_cpp }}
+    - cairo
     - curl
-    - freetype
-    - glib
-    - gobject-introspection 1.*  # [not win]
-    - jpeg
-    - libiconv                           # [osx]
-    - libpng
-    - libtiff
-    - libxcb                             # [linux]
-    - openjpeg >=2
-    - zlib
     - fontconfig
-    - libcurl
-    - gettext  # [osx]
-  run:
-    - poppler-data
-    - cairo  # [not win]
-    - curl
     - freetype
+    - gettext  # [osx]
     - glib
     - jpeg
+    - lcms2
+    - libcurl
+    - libiconv
     - libpng
     - libtiff
+    - nss  # [not (win or s390x)]
     - openjpeg
-    - fontconfig
-    - libcurl
+    - qt-main  # [not (ppc64le or s390x)]
     - zlib
-    - gettext  # [osx]
 
-test:
-  commands:
-    - pdfunite --help
-    - pdftocairo --help  # [not win]
+outputs:
+  - name: poppler
+    build:
+      missing_dso_whitelist:
+        - $RPATH/libm.so.6
+        - $RPATH/libc.so.6
+        - $RPATH/ld-linux-x86-64.so.2
+        - $RPATH/libpthread.so.0
+    script: install.sh   # [unix]
+    script: install.bat  # [not unix]
+    requirements:
+      build:
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+        # Need these CDTs for Qt on Linux. Please keep them alphabetized!
+        - {{ cdt('libselinux-devel') }}      # [linux]
+        - {{ cdt('libxdamage') }}            # [linux]
+        - {{ cdt('libxfixes') }}             # [linux]
+        - {{ cdt('libxscrnsaver-devel') }}   # [linux and not ppc64le]
+        - {{ cdt('libxtst-devel') }}         # [linux and not ppc64le]
+        - {{ cdt('libxxf86vm') }}            # [linux]
+        - {{ cdt('mesa-libgl-devel') }}      # [linux and not s390x]
+        - {{ cdt('mesa-libegl-devel') }}     # [linux and not ppc64le]
+        - {{ cdt('mesa-libgbm') }}           # [linux and not ppc64le]
+        - {{ cdt('mesa-dri-drivers') }}      # [linux]
+        - {{ cdt('libxau-devel') }}          # [linux]
+        - {{ cdt('libxext-devel') }}         # [linux]
+        - {{ cdt('libxrender-devel') }}      # [linux]
+        - {{ cdt('xorg-x11-proto-devel') }}  # [linux]
```

Comment:
Why do we need these CTDs for Qt on linux for the `poppler` output? Or do they require for `poppler-qt` only?

Reply:
because we need to have it on build-time so that X dependencies can be resolved in configure and testing